### ISSUE TITLE
Logic App Standard Not Replaced When App Service Plan Is Changed

### ISSUE THREAD

**Issue by JamieH-risual** [hashicorp/terraform-provider-azurerm#15030]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Latest on GitHub hosted runner 

### Affected Resource(s)

* `azurerm_logic_app_standard`

### Expected Behaviour

Resource should be replaced.

### Actual Behaviour

Apply step fails with service error:

`│ Error: updating Logic App Standard: (Site Name "** Logic App Name **" / Resource Group "** Resource Group Name **"): web.AppsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>`

### Steps to Reproduce

1. Create new app service plan and amend existing Logic App Standard to use the new plan.
2. `terraform apply`

### Additional Info

This also causes issues when making changes to the app service plan which requires it to be replaced. This is likely because the plan cannot be replaced while it hosts any apps. The apply fails with a similar service error:

`│ Error: deleting App Service Plan "** Plan Name **" (Resource Group "** Resource Group Name **"): web.AppServicePlansClient#Delete: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>`

**Comment by neil-yechenwei** [comment]
@JamieH-risual , thanks for raising this issue. After tested, I cannot repro this issue with below tf config. Could you try below tf config and latest azurerm provider to see if the issue still exists? Thanks.

The tf config for first apply:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-las-test01"
  location = "West Europe"
}

resource "azurerm_storage_account" "test" {
  name                     = "acctestsalastest01"
  resource_group_name      = azurerm_resource_group.test.name
  location                 = azurerm_resource_group.test.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_app_service_plan" "test" {
  name                = "acctestASP-test01"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  kind                = "elastic"

  sku {
    tier = "WorkflowStandard"
    size = "WS1"
  }
}

resource "azurerm_logic_app_standard" "test" {
  name                       = "acctest-las-test01"
  location                   = azurerm_resource_group.test.location
  resource_group_name        = azurerm_resource_group.test.name
  app_service_plan_id        = azurerm_app_service_plan.test.id
  storage_account_name       = azurerm_storage_account.test.name
  storage_account_access_key = azurerm_storage_account.test.primary_access_key
}
```

The tf config for second apply:
```
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "acctestRG-las-test01"
  location = "West Europe"
}

resource "azurerm_storage_account" "test" {
  name                     = "acctestsalastest01"
  resource_group_name      = azurerm_resource_group.test.name
  location                 = azurerm_resource_group.test.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_app_service_plan" "test" {
  name                = "acctestASP-test01"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  kind                = "elastic"

  sku {
    tier = "WorkflowStandard"
    size = "WS1"
  }
}

resource "azurerm_app_service_plan" "test2" {
  name                = "acctestASP-test02"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  kind                = "elastic"

  sku {
    tier = "WorkflowStandard"
    size = "WS1"
  }
}

resource "azurerm_logic_app_standard" "test" {
  name                       = "acctest-las-test01"
  location                   = azurerm_resource_group.test.location
  resource_group_name        = azurerm_resource_group.test.name
  app_service_plan_id        = azurerm_app_service_plan.test2.id
  storage_account_name       = azurerm_storage_account.test.name
  storage_account_access_key = azurerm_storage_account.test.primary_access_key
}
```

**Comment by ramonschopper** [comment]
I can confirm and reproduce the issue. Either change parameters on the App Service Plan which requires recreation or moving to a new App Service Plan resource will recognized as in-place update by Terraform plan:
` # azurerm_logic_app_standard.example will be updated in-place
  ~ resource "azurerm_logic_app_standard" "example" {
      ~ app_service_plan_id            = "/subscriptions/4c713b6e-2479-48bb-ab84-0b932c0a276b/resourceGroups/azure-functions-test-rg/providers/Microsoft.Web/serverfarms/example" -> (known after apply)
        id                             = "/subscriptions/4c713b6e-2479-48bb-ab84-0b932c0a276b/resourceGroups/azure-functions-test-rg/providers/Microsoft.Web/sites/test-azure-functions-rsc"
        name                           = "test-azure-functions-rsc"
        tags                           = {}
        # (20 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

  # azurerm_service_plan.example must be replaced
-/+ resource "azurerm_service_plan" "example" {
      ~ id                           = "/subscriptions/4c713b6e-2479-48bb-ab84-0b932c0a276b/resourceGroups/azure-functions-test-rg/providers/Microsoft.Web/serverfarms/example" -> (known after apply)
      ~ kind                         = "elastic" -> (known after apply)
      ~ maximum_elastic_worker_count = 1 -> (known after apply)
      ~ name                         = "example" -> "example2" # forces replacement
      ~ reserved                     = true -> (known after apply)
      - tags                         = {} -> null
      ~ worker_count                 = 1 -> (known after apply)
      - zone_balancing_enabled       = false -> null
        # (5 unchanged attributes hidden)
    }`

**Comment by saveler** [comment]
Hello everybody.
I haven't tried to reproduce it, but according to TF resource documentation, there is no information that changing app_service_plan_id forces resource recreation.

At the same time I opened source code of this resource: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/logic/logic_app_standard_resource.go

		Schema: map[string]*pluginsdk.Schema{
			"name": {
				Type:         pluginsdk.TypeString,
				Required:     true,
				ForceNew:     true,
				ValidateFunc: validate.LogicAppStandardName,
			},

			"resource_group_name": commonschema.ResourceGroupName(),

			"location": commonschema.Location(),

			"app_service_plan_id": {
				Type:     pluginsdk.TypeString,
				Required: true,
			},

			"app_settings": {
				Type:     pluginsdk.TypeMap,
				Optional: true,
				Computed: true,
				Elem: &pluginsdk.Schema{
					Type: pluginsdk.TypeString,
				},
			},

Here I can see that "name" parameter has flag "ForceNew: true". I think it is needed to tell terraform that resource has to be recreated once name of Logic App is changed, but there is no such flag for "app_service_plan_id" flag and it should be a reason.

**Comment by ramonschopper** [comment]
@saveler yes I went through the source of web app and functions and ForceNew is nowhere set.
